pulp_smash/tests/rpm/cli/test_upload_and_remove_langpacks.py

```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+"""Tests that perform langpacks upload and removal."""
+from __future__ import unicode_literals
+
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+
+class UploadAndRemoveLangpacksTestCase(unittest2.TestCase):
+    """Upload langpacks and remove it through pulp-rpm.
+
+    This test case targets `Pulp Smash #270`_. How do we test this?
+    The test steps would be as follows:
+
+    1. upload langpacks to repo1
+    2. check whether the langpacks are really uploaded
+    3. remove all langpacks from repo1
+    4. check whether the langpacks are actually removed
+
+    .. _Pulp Smash #270: https://github.com/PulpQE/pulp-smash/issues/270
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a repository."""
+        cls.cfg = config.get_config()
+        cls.client = cli.Client(cls.cfg)
+        cls.repo_id = utils.uuid4()
+        cls.client.run(
+            'pulp-admin rpm repo create --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+
+    def test_rpm_upload_and_remove(self):
+        """Upload and remove langpacks through pulp-rpm."""
+        # Upload langpacks into the repository
+        cmd = (
+            'pulp-admin rpm repo uploads langpacks -i hyphen '
+            '-n hyphen-%s --repo-id {}'
+        ).format(self.repo_id)
+        self.client.run(cmd.split())
+        package_counts = get_langpacks_counts_by_repo_id(
+            self.repo_id,
+            self.cfg
+        )
+        self.assertGreater(package_counts, 0, 'Langpacks are not uploaded')
+        # Remove all langpacks from the repository
+        cmd = (
+            'pulp-admin rpm repo remove langpacks --repo-id {0} '
+            '--str-eq repo_id={0}'
+        ).format(self.repo_id)
+        self.client.run(cmd.split())
+        package_counts = get_langpacks_counts_by_repo_id(
+            self.repo_id,
+            self.cfg
+        )
+        self.assertEqual(package_counts, 0, 'Langpacks are not removed')
+
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
Unfortunately, there is no common tear down machinery for the CLI. Thus far, I've been handling this by using base classes to minimize the amount of pain. :/